### PR TITLE
DDO: Refactoring, support additional format of public_key.id value

### PR DESCRIPTION
**Pull request by Patrik-Stas**
# Refactoring
- Separated DDO code to separate module
- Moved `Invitation -> DDO` conversions out of DDO next to invitation code
- Refactored DDO code, especially concerning references to `public_key` section, created type `DdoKeyReference`, modified function signatures to work with this type
- Renamed old ddo testing construction functions like `_did_doc_1` to have more descriptive names

# Features
- Added support to allow `id` in `public_key` objects to be in format `<key>#<key_id>` such as `Xnwiz15WsgFeN2CxQBuYo5#1`. Previously only format `key_id` such as `1` was allowed.

# Current DDO format support notes:
- we define `service` field as array of objects. These service objects reflect [aries specification](https://github.com/hyperledger/aries-rfcs/blob/main/features/0434-outofband/README.md#the-services-item) which is more rich than w3c specification for service [attribute on DDO](https://github.com/hyperledger/aries-rfcs/blob/main/features/0434-outofband/README.md#the-services-item)
- we, and seem that most other current Aries implementations define `publicKey` property on DDO, this is however not present in w3c specification. Arguably [`authentication` verification method](https://w3c.github.io/did-core/#did-document-properties) should be used instead and usage of `publicKey` should be dropped.

### CODE REVIEW

**Review by mirgee — CHANGES_REQUESTED** [review]

File: aries_vcx/src/did_doc/model.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+pub const KEY_AUTHENTICATION_TYPE: &str = "Ed25519SignatureAuthentication2018";
+
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
+pub struct Ed25519PublicKey {
+    pub id: String,
+    #[serde(rename = "type")]
+    pub type_: String,
+    // all list of types: https://w3c-ccg.github.io/ld-cryptosuite-registry/
+    pub controller: String,
+    #[serde(rename = "publicKeyBase58")] // todo: rename to publicKeyMultibase
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply aries_vcx/src/did_doc/model.rs:22]: created https://github.com/hyperledger/aries-vcx/issues/531

File: aries_vcx/src/did_doc/model.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    pub public_key_base_58: String,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, PartialEq)]
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply aries_vcx/src/did_doc/model.rs:26]: We do need these here as this is part of DDO, and these are properties which we want to have on DDO. I agree probably some derived impls across the codebase could be removed

File: aries_vcx/src/did_doc/model.rs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            None => serializer.collect_str(&self.key_id),
+            Some(did) => serializer.collect_str(&format!("{}#{}", did, self.key_id))
+        }
+    }
+}
+
+
+#[cfg(test)]
+#[cfg(feature = "general_test")]
+mod unit_test {
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply aries_vcx/src/did_doc/model.rs:55]: Agreed, let's consolidate this in separate PR

File: aries_vcx/src/handlers/service_resolvable.rs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+use crate::did_doc::service_aries::AriesService;
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply aries_vcx/src/handlers/service_resolvable.rs:1]: I wouldn't like to see it in `did_doc` - tried to keep that module clean of aries-vcx specifics as if we were to create independent crate out of it. 
I am think we can put it into `utils` though

**Review by mirgee — APPROVED** [review]

**Pull request merged**